tlog-mirror.md

```diff
@@ -0,0 +1,297 @@
+
+The base64 encoding used throughout is the standard Base 64 encoding specified
+in [RFC 4648][], Section 4.
+
+`U+` followed by four hexadecimal characters denotes a Unicode codepoint, to be
+encoded in UTF-8. `0x` followed by two hexadecimal characters denotes a byte
+value in the 0-255 range.
+
+`[start, end)`, where `start <= end`, denotes the half-open interval containing
+integers `x` such that `start <= x < end`.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in [BCP 14][] [RFC 2119][] [RFC
+8174][] when, and only when, they appear in all capitals, as shown here.
+
+[RFC 4648]: https://www.rfc-editor.org/rfc/rfc4648.html
+[BCP 14]: https://www.rfc-editor.org/info/bcp14
+[RFC 2119]: https://www.rfc-editor.org/rfc/rfc2119.html
+[RFC 8174]: https://www.rfc-editor.org/rfc/rfc8174.html
+[RFC 6962]: https://www.rfc-editor.org/rfc/rfc6962.html
+
+## Introduction
+
+A mirror is a [cosigner][] that stores a copy of a log and, when signing a
+[checkpoint][], provides the additional guarantee that the mirror has
+durably logged the contents of the checkpoint and has made it accessible.
+
+A mirror is defined by a name, a public key, and by two URL prefixes:
+the *submission prefix* for write APIs and the *monitoring prefix* for read
+APIs. A mirror MAY use the same value for both the *submission prefix* and the
+*monitoring prefix*.
+
+For each supported origin log, the mirror is configured with:
+
+* The log's public key
+* The log's URL prefix
+* An optional list of monitoring prefixes for other mirrors for the log
+
+The mirror maintains a copy of each origin log and serves it publicly via the
+[tiled transparency log][] interface. It uses a URL prefix of
+`<monitoring prefix>/<encoded origin>`, where `encoded origin` is the log's
+origin, [percent-encoded][]. The checkpoint served from this prefix MUST include
+a [cosignature][] from the mirror.
+
+## Updating a Mirror
+
+The mirror update process is designed to be safely interruptible, while avoding
+large atomic operations. For each origin log, a mirror maintains the following:
+
+* A copy of the log. The latest checkpoint of this log copy is known as the
+  *mirror checkpoint*.
+
+* A *pending checkpoint*, which is at or ahead of the mirror checkpoint. If
+  ahead of the mirror checkpoint, the pending checkpoint describes entries that
+  have yet to be incorporated into the mirror.
+
+* A list of *pending entries* that have yet to be incorporated into the mirror
+  checkpoint. The mirror's *next entry* is the log index of the first entry that
+  is not in either the log or pending entry list.
+
+The update process ensures that all current and past pending checkpoints are
+consistent, and all pending entries are contained in the current pending
+checkpoint. Thus mirrors MAY commit pending entries to the log, serving them as
```

Comment:
I'd missed that the mirror could go ahead and serve pending entry bundles before the corresponding mirror checkpoint is updated. Makes sense!

Reply:
The text seems clear enough now!